fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/api/trading/v1/TradeOfferFactory.java

```diff
@@ -0,0 +1,123 @@
+	}
+
+	/**
+	 * Represents a trade where a villager buys an item and for a few emeralds, will "process" the item.
+	 *
+	 * <p>This is commonly used for cooking items (6 emeralds + 8 raw pork -> 8 cooked pork).
+	 * @param itemConvertible The item to be processed
+	 * @param secondCount The required amount of the item being processed.
+	 * @param price The required amount of emeralds.
+	 * @param sellItem The item which is sold back to the player.
+	 * @param sellCount The amount of the item sold back to the player.
+	 * @param maxUses The amount of "uses" a trade can take before a villager needs to restock.
+	 * @param experience The amount of experience the {@link VillagerEntity Villager} will receive.
+	 * @return A new trade offer factory.
+	 */
+	TradeOffers.Factory createProcessItemFactory(ItemConvertible itemConvertible, int secondCount, int price, Item sellItem, int sellCount, int maxUses, int experience);
+
+	/**
+	 * Represents a trade offer where a villager will sell an item or amount of items for some emeralds.
+	 * @param item The item this villager will sell.
+	 * @param price The amount of emeralds needed to buy this item.
+	 * @param count The amount of the item sold.
+	 * @param maxUses The amount of "uses" a trade can take before a villager needs to restock.
+	 * @param experience The amount of experience the {@link VillagerEntity Villager} will receive.
+	 * @return A new trade offer factory.
+	 */
+	default TradeOffers.Factory createSellItemFactory(Item item, int price, int count, int maxUses, int experience) {
+		return this.createSellItemFactory(new ItemStack(item), price, count, maxUses, experience);
+	}
+
+	/**
+	 * Represents a trade offer where a villager will sell an item or amount of items for some emeralds.
+	 * @param itemStack The item this villager will sell.
+	 * @param price The amount of emeralds needed to buy this item.
+	 * @param count The amount of the item sold.
+	 * @param maxUses The amount of "uses" a trade can take before a villager needs to restock.
+	 * @param experience The amount of experience the {@link VillagerEntity Villager} will receive.
+	 * @return A new trade offer factory.
+	 */
+	default TradeOffers.Factory createSellItemFactory(ItemStack itemStack, int price, int count, int maxUses, int experience) {
+		return this.createSellItemFactory(itemStack, price, count, maxUses, experience, 0.05F);
+	}
+
+	/**
+	 * Represents a trade offer where a villager will sell an item or amount of items for some emeralds.
+	 * @param itemStack The item this villager will sell.
+	 * @param price The amount of emeralds needed to buy this item.
+	 * @param count The amount of the item sold.
+	 * @param maxUses The amount of "uses" a trade can take before a villager needs to restock.
+	 * @param experience The amount of experience the {@link VillagerEntity Villager} will receive.
+	 * @param multiplier A price multiplier which represents a possible deviation from the standard price.
```

Comment:
This is like the rate of price growth after usage of this offer

Reply:
To be fair I wouldn't have known that since it's pretty vague in the mappings, maybe some TradeOffer javadoc in the future could be nice.